uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/FloorFilterController.qml

```diff
@@ -0,0 +1,703 @@
+
+    /*!
+      \brief Sets the \a facilityId as \l selectedFacilityId.
+      This will load the \l levels with the levels withing the facility and
+      reset by selecting \l selectedLevelId with level matching vertical ordering == 0.
+      Set empty string to reset the facility.
+      \sa setSelectedLevelId
+    */
+    function setSelectedFacilityId(facilityId) {
+        // avoid findind null idx, immediately return if empty string
+        if(facilityId === ""){
+            internal.selectedFacilityId = facilityId;
+            return;
+        }
+
+        let idx = findElementIdxById(facilityId,
+                                     FloorFilterController.TypeElement.Facility);
+        if (idx == null) {
+            console.error("not found facility");
+            internal.selectedFacilityId = "";
+            return;
+        }
+
+        // check that the current site is the selected facility selected's parent site. Otherwise the click came from a populateAllFacilities
+        var facility = floorManager.facilities[idx];
+
+        if (facility.site != null && (!internal.selectedSite || facility.site.siteId !== internal.selectedSite.siteId) ) {
+            console.log(facility.site === undefined, facility.site === null);
+            // selection of facility came after click of populateAllFacilities.
+            // this also sets the internal.selectedSite
+            internal.selectedSiteId = facility.site.siteId;
+        }
+        internal.selectedFacilityId = facilityId;
+    }
+
+    /*!
+      \qmlproperty string selectedSiteId
+      \brief Stores the current selected site Id.
+      Default is {""}.
+    */
+    readonly property alias selectedSiteId: internal.selectedSiteId
+
+    /*!
+      \brief Sets the \a siteId as \l selectedSiteId.
+      This will load the \l facilities with the facilities within the site and reset \l selectedFacilityId.
+      Does not repopulate facilities in case of not \l selectedSiteRespected.
+    */
+    function setSelectedSiteId(siteId) {
+        internal.selectedSiteId = siteId;
+    }
+
+    /*!
+      \qmlproperty ListModel levels
+      \brief Stores the levels in the currently selected facility in alphabetical order.
+      If not \l selectedSiteRespected, all levels are stored.
+      \sa selectedLevelId
+    */
+    property ListModel levels: ListModel {}
+
+    /*!
+      \qmlproperty ListModel facilities
+      \brief Stores the facilities in the currently selected site.
+      \sa selectedSiteId
+    */
+    property ListModel facilities: ListModel {}
+
+    /*!
+      \qmlproperty ListModel sites
+      \brief Stores all the sites in the current \l floorManager.
+      \sa selectedSiteId
+    */
+    property ListModel sites: ListModel {}
+
+    property alias selectedSite: internal.selectedSite
+    property alias selectedFacility: internal.selectedFacility
+    property alias selectedLevel: internal.selectedLevel
```

Comment:
Should these be `readonly`, are they allowed to be settable?

Reply:
No, they should not be settable directly from user. Just reflect what is in the `internal.selected<>`